sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareClientBuilder.java

```diff
@@ -208,13 +208,17 @@ public ShareAsyncClient buildAsyncClient() {
         }, retryOptions, coreRetryOptions, logOptions, clientOptions, httpClient,
             perCallPolicies, perRetryPolicies, configuration, LOGGER);
 
+        if (sasToken != null && azureSasCredential == null) {
```

Comment:
I thought we decided we could remove this because we can get rid of the sasToken field?